src/Resources/skeleton/doctrine/Repository-use-composition.tpl.php

```diff
@@ -0,0 +1,96 @@
+<?= "<?php\n"; ?>
+
+namespace <?= $namespace; ?>;
+
+use <?= $entity_full_class_name; ?>;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+<?= $with_password_upgrade ? "use Symfony\Component\Security\Core\Exception\UnsupportedUserException;\n" : '' ?>
+<?= ($with_password_upgrade && str_contains($password_upgrade_user_interface->getFullName(), 'Password')) ? sprintf("use %s;\n", $password_upgrade_user_interface->getFullName()) : null ?>
+<?= $with_password_upgrade ? "use Symfony\Component\Security\Core\User\PasswordUpgraderInterface;\n" : '' ?>
+<?= ($with_password_upgrade && str_contains($password_upgrade_user_interface->getFullName(), '\UserInterface')) ? sprintf("use %s;\n", $password_upgrade_user_interface->getFullName()) : null ?>
+
+final class <?= $class_name; ?><?= $with_password_upgrade ? " implements PasswordUpgraderInterface\n" : "\n" ?>
+{
+    private EntityRepository $repository;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->repository = $entityManager->getRepository(<?= $entity_class_name; ?>::class);
+    }
+
+    public function find($id, $lockMode = null, $lockVersion = null): ?<?= $entity_class_name; ?><?= "\n"?>
+    {
+        return $this->repository->find($id, $lockMode, $lockVersion);
+    }
+
+    public function findOneBy(array $criteria, array $orderBy = null): ?<?= $entity_class_name; ?><?= "\n"?>
```

Comment:
I'm wondering if it would make more sense to have these methods in a `RepositoryTrait::class` to keep the end users code a dry.. Personally I'm not a fan of have lengthy generated PHP files that are duplicated in a code base.

Reply:
One of the purpose of this change is to have a strict typing, in PHP and no more in PHPDoc.

If we use a trait, there will be no more typing because the trait will be common to all repositories. And thus pretty useless.